### PR TITLE
wait for livy server socket to be in listening state before running test cases

### PR DESCRIPTION
**Pull request by yl790**
I may have figured out why the CI test script frequently fails with "Failed to launch livy session, session status is dead" even after Livy is installed successfully.

Let's see if this change helps.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -96,6 +96,7 @@ matrix:
         apt:
           packages:
             - openjdk-8-jre
+            - netcat
```
Comment (yl790) [inline .travis.yml]: actually won't need netcat anywhere... gonna wait for CI to complete, and then delete this and run CI again

**Comment by javierluraschi** [comment]
@yitao-li we could also consider moving some tests to use GitHub actions... they run sooooo much faster than Travis and we've had good success in other projects like https://github.com/mlverse/lantern

**Comment by javierluraschi** [comment]
BTW. Even if this doesn't fix the test, I think is a good change to merge, as in, just in case.

**Comment by javierluraschi** [comment]
We might be able to get meaningful error information if we change stdout to `TRUE` here:

https://github.com/sparklyr/sparklyr/blob/master/tests/testthat/helper-initialize.R#L201-L205

We can also consider adding Livy tests to allowed failures:

https://github.com/sparklyr/sparklyr/blob/master/.travis.yml#L109-L123

**Comment by javierluraschi** [comment]
My guess is that when a PR gets created, Travis uses machines with less memory, where Livy fails to start due to low memory or something around those lines.

**Pull request merged**